app/src/lib/stores/app-store.ts

```diff
@@ -2887,26 +2899,32 @@ export class AppStore extends TypedBaseStore<IAppState> {
             repository,
             branch,
           },
+          gitContext: { kind: 'checkout', branchToCheckout: foundBranch.name },
         }
       )
     )
 
-    if (shouldPopStash) {
-      const stash = await getLastDesktopStashEntryForBranch(
-        repository,
-        foundBranch.name
-      )
-
-      if (stash !== null) {
+    if (
```

Comment:
Are we concerned here about proceeding with this flow if `checkoutBranch` fails? This is handled in our Git error handling, but we don't poke at the result returned from `performFailableOperation`?

Reply:
I'm going to put a fix in place for now with the intent to address it next week when we pair on simplifying this function a bit.